src/main/java/org/bahmni/reports/template/TSIntegrationDiagnosisReportTemplate.java

```diff
@@ -0,0 +1,145 @@
+import org.apache.logging.log4j.Logger;
+import org.bahmni.reports.model.Report;
+import org.bahmni.reports.model.TSIntegrationDiagnosisReportConfig;
+import org.bahmni.reports.model.TSPageObject;
+import org.bahmni.reports.model.UsingDatasource;
+import org.bahmni.reports.report.BahmniReportBuilder;
+import org.bahmni.reports.util.CommonComponents;
+import org.bahmni.webclients.HttpClient;
+import org.stringtemplate.v4.ST;
+
+import java.io.IOException;
+import java.net.URI;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Properties;
+
+import static net.sf.dynamicreports.report.builder.DynamicReports.col;
+import static net.sf.dynamicreports.report.builder.DynamicReports.type;
+import static org.bahmni.reports.template.Templates.minimalColumnStyle;
+import static org.bahmni.reports.util.FileReaderUtil.getFileContent;
+
+
+@UsingDatasource("openmrs")
+public class TSIntegrationDiagnosisReportTemplate extends BaseReportTemplate<TSIntegrationDiagnosisReportConfig> {
+    public static final String CREATE_SQL_TEMPLATE = "create temporary table {0}(code varchar(100) not null)";
+    public static final String INSERT_SQL_TEMPLATE = "insert into {0} values (?)";
+
+    private static final Logger logger = LogManager.getLogger(TSIntegrationDiagnosisReportTemplate.class);
+    private HttpClient httpClient;
+    private Properties tsProperties;
+    private String descendantsUrlTemplate;
+
+    public TSIntegrationDiagnosisReportTemplate(HttpClient httpClient, Properties tsProperties, String descendantsUrlTemplate) {
+        super();
+        this.httpClient = httpClient;
+        this.tsProperties = tsProperties;
+        this.descendantsUrlTemplate = descendantsUrlTemplate;
+    }
+
+    public void setDescendantsUrlTemplate(String descendantsUrlTemplate) {
+        this.descendantsUrlTemplate = descendantsUrlTemplate;
+    }
+
+    @Override
+    public BahmniReportBuilder build(Connection connection, JasperReportBuilder jasperReport, Report<TSIntegrationDiagnosisReportConfig> report, String
+            startDate, String endDate, List<AutoCloseable> resources, PageType pageType) {
+        String tempTableName = "tmpCodes_" + System.nanoTime();
+        loadTempTable(connection, tempTableName, report.getConfig().getTerminologyParentCode());
+        String sql = getFileContent("sql/tsIntegrationDiagnosisCount.sql");
+
+        CommonComponents.addTo(jasperReport, report, pageType);
+        jasperReport.addColumn(col.column("Diagnosis", "Diagnosis", type.stringType()).setStyle(minimalColumnStyle).setHorizontalAlignment(HorizontalAlignment.CENTER));
+        if (report.getConfig().isDisplayTerminologyCode()) {
+            jasperReport.addColumn(col.column("Terminology Code", "Terminology Code", type.stringType()).setStyle(minimalColumnStyle).setHorizontalAlignment(HorizontalAlignment.CENTER));
```

Comment:
The column name needs to be a configuration which a user can specify. Not hardcoded.

Reply:
we have made the "Terminology Code" as configurable from snomed-default-config